ttnn/cpp/ttnn/tensor/layout/tensor_layout.cpp

```diff
@@ -0,0 +1,200 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "tensor_layout.hpp"
+
+namespace tt::tt_metal {
+
+namespace {
+size_t round_up(size_t value, size_t multiple) {
+    TT_FATAL(multiple != 0, "round_up: multiple must not be 0");
+
+    // can be faster if multiple is power of 2
+    // return (value + multiple - 1) & ~(multiple - 1);
+    return ((value + multiple - 1) / multiple) * multiple;
+};
+
+ttnn::Alignment legacyPaddedShapeToAlignment(const ttnn::SimpleShape& legacy_padded_shape) {
+    const auto rank = legacy_padded_shape.rank();
+    ttnn::SmallVector<uint32_t> values(rank);
+
+    if(rank >= 1) {
+        values[rank - 1] = legacy_padded_shape[rank - 1];
+    }
+    if(rank >= 2) {
+        values[rank - 2] = legacy_padded_shape[rank - 2];
+    }
+    for (int i = rank - 3; i >= 0; i--) {
```

Comment:
rank might be unsigned, be careful with subtracting from it

Reply:
Yep, mindful of that! Thank you!